client/src/Helper.ts

```diff
@@ -102,4 +103,38 @@ export class Helper {
         if (!progress) return "0%";
         return progress.toFixed(0) + "%";
     }
+
+    /**
+     * Returns the path to the global storage location provided by VSCode to the extension
+     */
+    public static getGlobalStoragePath(context: vscode.ExtensionContext): string {
+        return context.globalStorageUri.fsPath;
+    }
+
+    /**
+     * Returns the directory in which log files should be stored.
+     * The directory will be created if it does not exist yet
+     */
+     public static getLogDir(): string {
+        // check if a particular dir has been passed as an environment variable.
+        // this is mainly used for CI purposes:
+        let logDir = process.env["VIPER_IDE_LOG_DIR"];
+        if (logDir == null || logDir === "") {
+            logDir = path.join(os.tmpdir(), ".vscode");
+        }
+        //create logfile's directory if it wasn't created before
+        if (!fs.existsSync(logDir)) {
+            fs.mkdirSync(logDir, { recursive: true });
+        }
+        return logDir;
+    }
+
+    /**
+     * Returns true if `getGobraToolsPath` should be wiped after activating the extension to ensure a clean system state.
+     */
+    public static cleanInstall(): boolean {
+        const value = process.env["VIPER_IDE_CLEAN_INSTALL"];
+        return value != null &&

```

Comment:
I use `==` on purpose for comparison with `null` as recommended by e.g. Google: https://google.github.io/styleguide/jsguide.html#features-equality-checks

Reply:
Yes, I understand that coercion allows you to catch ```null``` and ```undefined``` using the same condition. But since we use TypeScript, it seems like a better option to avoid nullable types in the first place. At least theoretically, if you always write e.g. ```var x: Type | undefined = undefined```, then you can always check ```x === undefined```. Otherwise, the ```==``` for checking null values to me feels like an ugly special case. 

However, I don't know what the specification of ```process.env``` is, and I certainly don't insist on changing anything, these were just my thoughts. :-)